app/controllers/users.js

```diff
@@ -0,0 +1,19 @@
+const logger = require('../logger');
+const usersService = require('../services/users');
+const signUpValidator = require('../interactors/sign_up_validator');
+
+module.exports.signUp = (req, res, next) => {
+  try {
```

Comment:
This is sync now, so i understand that we need a try/catch block to catch the exception.
Also i checked this like the previous way => 
try / catch(e) { next(e) } 
and didn't work, it keep creating the user. Because of that, this change. Now works

Reply:
This still has the same problem of not forwarding the error to the errors handler. `try / catch(e) { next(e) }` is fine, you just need to `return` the `next`, in order to cut the flow there and prevent the rest of the controller from executing.